lib/core/mqtt.cpp

```diff
@@ -0,0 +1,161 @@
+#include "mqtt.hpp"
+
+#include <rapidjson/writer.h>
+
+#include "rapidjson/stringbuffer.h"
+
+namespace hyped::core {
+
+std::optional<std::shared_ptr<Mqtt>> Mqtt::create(ILogger &logger,
+                                                  const std::string &id,
+                                                  const std::string &host,
+                                                  const uint16_t port)
+{
+  std::string address = "tcp://" + host + ":" + std::to_string(port);
+  mqtt::connect_options connection_options;
+  connection_options.set_keep_alive_interval(1);
+  connection_options.set_clean_session(true);
+  auto mqtt_client = std::make_unique<mqtt::client>(address, id);
+  mqtt_client->connect(connection_options);
+  if (!mqtt_client->is_connected()) { return std::nullopt; }
+  return std::make_shared<Mqtt>(logger, std::move(mqtt_client));
+}
+
+Mqtt::Mqtt(ILogger &logger, std::unique_ptr<mqtt::client> client)
+    : logger_(logger),
+      client_(std::move(client)),
+      messages_in_queue(0)
+{
+}
+
+Mqtt::~Mqtt()
+{
+  client_->disconnect();
+}
+
+void Mqtt::publish(const MqttMessage &message, const MqttMessageQos qos)
+{
+  auto constructed_message = messageToMessagePtr(message);
+  logger_.log(
+    core::LogLevel::kDebug, "Publishing message %s", constructed_message->get_payload_str());
```

Comment:
This should fix the dodgy logs
```suggestion
    core::LogLevel::kDebug, "Publishing message %s", constructed_message->get_payload_str().c_str());
```
`get_payload_str()` returns an `mqtt::string`, which apparently isn't printable (why is there a seperate def of string in paho??)